src/query/server/query.go

```diff
@@ -504,6 +528,10 @@ func newM3DBStorage(
 			if err != nil {
 				return nil, nil, nil, nil, err
 			}
+
+			if downsamplerReadyCh != nil {
+				downsamplerReadyCh <- struct{}{}
+			}
```

Comment:
delete this? should get done in the defer right?

Reply:
This is in an else clause - defer is inside an inner function inside the other side of the else.

Reply:
(I think)

Reply:
Ah, then won’t you miss a send when the newDownsamplerFn errors on this branch? Can we defer before this if else to avoid this? Could see this being a bit dangerous going forward

Reply:
I'll move it into `newDownsamplerFn` so it's only done in one place. 

I can't defer from the top level function since some of these are async clients and are not ready by the end of the top level function creating these async clients.

For some further context`DownsamplerReadyCh` is only used by tests to ensure failures to write a datapoint doesn't fail due to the async downsampler being not ready. The `DownsamplerReadyCh` should not be notified unless the downsampler is truly ready, if an error occurs then the channel should never receive an update.